cmd/user-cluster-controller-manager/main.go

```diff
@@ -433,15 +435,24 @@ func main() {
 		if err != nil {
 			log.Fatalw("Failed to get KubeVirt infra kubeconfig", zap.Error(err))
 		}
+		// VM and VMIs are created in a namespace having the same name as the cluster namespace name.
+		kvCacheOpts := cache.Options{
+			DefaultNamespaces: map[string]cache.Config{
+				runOp.namespace: {},
+			},
+		}
+		// If kv-infra-namespace flag is set, that means Kubevirt is running in the namespaced mode, so all workloads will be deployed in the specified namespace.
+		if runOp.kubeVirtInfraNamespace != "" {
+			kvCacheOpts = cache.Options{
+				DefaultNamespaces: map[string]cache.Config{
+					runOp.kubeVirtInfraNamespace: {},
+				},
+			}
+		}
```

Comment:
Personally I would avoid all this structs in structs and structs with maps and reduce this to the actual difference:

```
namespace := runOp.namespace
// If kv-infra-namespace flag is set, that means Kubevirt is running in the namespaced mode, so all workloads will be deployed in the specified namespace.
if kvNamespace := runOp.kubeVirtInfraNamespace; kvNamespace != "" {
  namespace = kvNamespace
}

manager.New(kubevirtInfraConfig, manager.Options{
			LeaderElection: false,
			Metrics:        metricsserver.Options{BindAddress: "0"},
			Cache: cache.Options{
				DefaultNamespaces: map[string]cache.Config{
					namespace: {},
				},
			},
		})
```

Reply:
agree